ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java

```diff
@@ -295,12 +307,15 @@ public static void main(String[] args) {
    * @throws IOException
    * @throws GeneralSecurityException
    */
-  private NettyPerfClient(String host, int port, String path, int concurrency, Long totalSize, Integer chunkSize,
-      String sslPropsFilePath, String serviceId, String targetAccountName, String targetContainerName,
-      List<String> customHeaders) throws Exception {
+  private NettyPerfClient(String host, int port, String path, String pathFileName, int concurrency, Long totalSize,
+      Integer chunkSize, String sslPropsFilePath, String serviceId, String targetAccountName,
+      String targetContainerName, List<String> customHeaders) throws Exception {
     this.host = host;
     this.port = port;
     this.path = path;
+    if (pathFileName != null) {
+      this.pathList = Files.readAllLines(Paths.get(pathFileName));
```

Comment:
Each blobId is 78 bytes. 1GB memory can have 12820512 blobIds. Based on the fact that average blob size is 120KB. 1GB memory can be used to GET 1.5TB data which is big enough for testing.